libnetwork/drivers/overlay/ov_network.go

```diff
@@ -639,32 +587,62 @@ func (n *network) setupSubnetSandbox(s *subnet, brName, vxlanName string) error
 		return fmt.Errorf("vxlan interface creation failed for subnet %q: %v", s.subnetIP.String(), err)
 	}
 
-	if !hostMode {
-		var name string
-		for _, i := range sbox.Info().Interfaces() {
-			if i.Bridge() {
-				name = i.DstName()
-			}
+	if hostMode {
+		return addFilters(n.id[:12], brName)
+	}
+
+	if err := setDefaultVLAN(sbox); err != nil {
+		// not a fatal error
+		logrus.WithError(err).Error("set bridge default vlan failed")
+	}
+	return nil
+}
+
+func setDefaultVLAN(sbox osl.Sandbox) error {
+	var brName string
+	for _, i := range sbox.Info().Interfaces() {
+		if i.Bridge() {
+			brName = i.DstName()
 		}
-		cmd := &exec.Cmd{
-			Path:   reexec.Self(),
-			Args:   []string{"set-default-vlan", sbox.Key(), name},
-			Stdout: os.Stdout,
-			Stderr: os.Stderr,
+	}
+
+	var innerErr error
+	err := sbox.InvokeFunc(func() {
+		// Mount sysfs, making sure the sysfs mount doesn't propagate
+		// back. The rationale for doing so has been lost to the mists
```

Comment:
The reason this needs to be remounted is most likely to ensure that /sys/class/net has the correct view of the network.
Without remounting you still see the host's view.

You can try this out with `unshare -n`

Reply:
Similar needs to be done for /proc for pid namespaces.

Reply:
You're absolutely right! I had initially thought that might have been the reason, but I dismissed that without testing (until today) because the sysfs(5) man page says

> /sys/class/net
> Each of the entries in this directory is a symbolic link
> representing one of the real or virtual networking devices
> that are visible **in the network namespace of the process
> that is accessing the directory.**  Each of these symbolic
> links refers to entries in the /sys/devices directory.